src/solver/optimisation/adequacy_patch.cpp

```diff
@@ -27,49 +27,107 @@
 
 #include "../simulation/simulation.h"
 #include "adequacy_patch.h"
+#include <map>
 
-using namespace Antares;
-using namespace Antares::Data;
-using namespace Antares::Data::AdequacyPatch;
+namespace Antares
+{
+namespace Data
+{
+namespace AdequacyPatch
+{
+AdequacyPatch::LinkCapacityForAdequacyPatchFirstStep SetNTCForAdequacyFirstStep(
+  AdequacyPatch::AdequacyPatchMode OriginNodeAdequacyPatchType,
+  AdequacyPatch::AdequacyPatchMode ExtremityNodeAdequacyPatchType,
+  std::map<adqPair, LinkCapacityForAdequacyPatchFirstStep>& behaviorMap)
+{
+    auto search = behaviorMap.find(
+      std::make_pair(OriginNodeAdequacyPatchType, ExtremityNodeAdequacyPatchType));
+    if (search != behaviorMap.end())
+        return search->second;
+    else
+        return leaveLocalValues;
+}
 
-NTC SetNTCForAdequacyFirstStep(bool AdequacyFirstStep,
-                               AdequacyPatchMode StartNodeAdequacyPatchType,
-                               AdequacyPatchMode EndNodeAdequacyPatchType,
-                               bool SetToZero12LinksForAdequacyPatch,
-                               bool SetToZero11LinksForAdequacyPatch)
+std::map<adqPair, LinkCapacityForAdequacyPatchFirstStep> GenerateLinkRestrictionMapForAdqFirstStep(
+  bool SetToZero12LinksForAdequacyPatch,
+  bool SetToZero11LinksForAdequacyPatch)
 {
-    if (AdequacyFirstStep)
+    std::map<adqPair, LinkCapacityForAdequacyPatchFirstStep> behaviorMap;
+
+    behaviorMap.insert(std::make_pair(
+      std::make_pair(adqmPhysicalAreaInsideAdqPatch, adqmPhysicalAreaInsideAdqPatch), setToZero));
+    behaviorMap.insert(std::make_pair(
+      std::make_pair(adqmPhysicalAreaOutsideAdqPatch, adqmPhysicalAreaInsideAdqPatch),
+      (SetToZero12LinksForAdequacyPatch) ? setToZero : setExtremityOrigineToZero));
+    behaviorMap.insert(std::make_pair(
+      std::make_pair(adqmPhysicalAreaInsideAdqPatch, adqmPhysicalAreaOutsideAdqPatch),
+      (SetToZero12LinksForAdequacyPatch) ? setToZero : setOrigineExtremityToZero));
+    behaviorMap.insert(std::make_pair(
+      std::make_pair(adqmPhysicalAreaOutsideAdqPatch, adqmPhysicalAreaOutsideAdqPatch),
+      (SetToZero11LinksForAdequacyPatch) ? setToZero : leaveLocalValues));
+    behaviorMap.insert(
+      std::make_pair(std::make_pair(adqmVirtualArea, adqmVirtualArea), leaveLocalValues));
+    behaviorMap.insert(std::make_pair(
+      std::make_pair(adqmVirtualArea, adqmPhysicalAreaOutsideAdqPatch), leaveLocalValues));
+    behaviorMap.insert(std::make_pair(
+      std::make_pair(adqmPhysicalAreaOutsideAdqPatch, adqmVirtualArea), leaveLocalValues));
+    behaviorMap.insert(std::make_pair(
+      std::make_pair(adqmVirtualArea, adqmPhysicalAreaInsideAdqPatch), leaveLocalValues));
+    behaviorMap.insert(std::make_pair(
+      std::make_pair(adqmPhysicalAreaInsideAdqPatch, adqmVirtualArea), leaveLocalValues));
+
+    return behaviorMap;
```

Comment:
I do not think this is more clear than the previous version.

Reply:
I tried to make it clearer with a few comments and putting the conditional check outside.